SQL statements/UpdateBestsellers()_PROCEDURE.sql

```diff
@@ -0,0 +1,56 @@
+ALTER TABLE BOOKS ADD BESTSELLER BOOLEAN;
+
+DROP FUNCTION IF EXISTS isBestSeller;
+
+DELIMITER $$
+
+CREATE FUNCTION isBestSeller(avgbookrents DECIMAL(5,2)) RETURNS BOOLEAN DETERMINISTIC
+BEGIN
+    DECLARE bestseller BOOLEAN;
+    IF avgbookrents > 2 THEN
+        SET bestseller = true;
+    ELSE
+        SET bestseller = false;
+    END IF;
+    RETURN bestseller;
+END $$
+
+DELIMITER ;
+
+DROP PROCEDURE IF EXISTS UpdateBestsellers;
+
+DELIMITER $$
+
+CREATE PROCEDURE UpdateBestsellers()
+BEGIN
+    DECLARE BOOKRENTS, DAYS, THIS_BOOK_ID INT;
+    DECLARE RENTSPERMONTH DECIMAL(5,2);
+    DECLARE FINISHED INT DEFAULT 0;
+    DECLARE ALL_BOOKS CURSOR FOR SELECT BOOK_ID FROM BOOKS;
+    DECLARE CONTINUE HANDLER FOR NOT FOUND SET FINISHED = 1;
+    OPEN ALL_BOOKS;
+    WHILE (FINISHED = 0) DO
+            FETCH ALL_BOOKS INTO THIS_BOOK_ID;
+            IF (FINISHED = 0) THEN
+                SELECT COUNT(*) FROM RENTS
+                    WHERE BOOK_ID = THIS_BOOK_ID
+                        INTO BOOKRENTS;
+                SELECT DATEDIFF(MAX(RENT_DATE), MIN(RENT_DATE)) + 1 FROM RENTS
+                    WHERE BOOK_ID = THIS_BOOK_ID
+                        INTO DAYS;
+                IF DAYS < 30 THEN
```

Comment:
a co to?

Reply:
Wyliczam średnią ilość wypożyczeń danej książki w miesiącu. Tam był podany taki wzór wyliczenia tj. odjąć datę ostatniego wypożyczenia od pierwszego wypożyczenia dzielone przez 30. Tylko ja jeszcze dodałem if'a, że jeśli ten czas jest mniejszy niż 30 dni to średnia ilość niech wynosi po prostu tyle ile było wypożyczeń w ostatnim miesiącu. Bo inaczej wychodzą dziwne liczby np. książka została wypożyczona tylko 3 razy, a średnia wypożyczeń to 15 razy na miesiąc

Reply:
Okej, niech tak zostanie :) Ja bym do tego podszedł inaczej